auth/gcloud/aio/auth/session.py

```diff
@@ -0,0 +1,157 @@
+import threading
+from abc import ABCMeta
+from abc import abstractmethod
+from abc import abstractproperty
+from io import IOBase
+from typing import Any
+from typing import Dict
+
+import requests
+
+from .build_constants import BUILD_GCLOUD_REST
+
+class BaseSession():
+    __metaclass__ = ABCMeta
+
+    def __init__(self, session=None, conn_timeout: int = 10,
```

Comment:
Untyped `session` in typed context?

Reply:
Also no return types?

Reply:
Hm, I would have thought the base class would work, right? Unless `mypy` doesn't work properly (I wouldn't be surprised).

Reply:
Well, to be honest, I don't know, but I specifically did not put a typing hint here because of differing types of `session` in `AioSession` and `SyncSession`. I can put something but I am not sure if it will actually help.

Do you have a suggestion on how we might improve this?

Reply:
Ah, for some reason I thought this was related to the session objects you were defining, but I misread the code. I think it's fine as is for now!